### PR TITLE
Fix delete button style

### LINKED ISSUE #9050: DeleteButton styling
**What you were expecting:**
When using `<DeleteButton/>` I'd expect it to use the palette defined in my custom theme, especially `color` and `contrastText`.

```
  <DeleteButton
    variant="contained"
    color="primary"
    label="Delete"
  />
```
I'd expect to have : 
image

**What happened instead:**

With the upper code I end up with this :

image

**Steps to reproduce:**

With the following theme :
```ts
export const theme = {
  ...defaultTheme,
  palette: {
    primary: {
      light: colors.orange[100],
      main: colors.orange[500],
      contrastText: colors.anthracite[50],
    },
    error: {
      main: colors.orange[500],
    },
  },
};
```

Add the DeleteButton like in the upper code.

**Related code:**

https://stackblitz.com/edit/github-uos7fp?file=src%2Fcomments%2FCommentEdit.tsx
https://stackblitz.com/edit/github-uos7fp?file=src%2Ftheme.ts

**Other information:**

**Environment**

* React-admin version: 4.11.4
* React version: 18.2.0
* Browser: [Version 1.52.126 Chromium: 114.0.5735.133 (Build officiel) (64 bits)](https://brave.com/latest/)

**Other information:**

I have a workaround :
```tsx
<DeleteButton
  variant="contained"
  color="primary"
  sx={{ color: colors.anthracite[50], '&:hover': { color: colors.orange[500] } }} />
```

### PR DESCRIPTION
**Pull request by adguernier**
Fix #9050

### CODE REVIEW

File: packages/ra-ui-materialui/src/button/DeleteWithConfirmButton.tsx [hunk 1]
```diff
@@ -131,19 +133,3 @@ DeleteWithConfirmButton.propTypes = {
...
-})(({ theme }) => ({
-    color: theme.palette.error.main,
-    '&:hover': {
-        backgroundColor: alpha(theme.palette.error.main, 0.12),
-        // Reset on mouse devices
-        '@media (hover: none)': {
-            backgroundColor: 'transparent',
-        },
-    },
-}));
```
Comment (slax57) [inline packages/ra-ui-materialui/src/button/DeleteWithConfirmButton.tsx:149]: I couldn't find any case where the removal of this code would have any impact, but I'm still wondering why we had it before.
@djhi  Do you have any clues? Can we remove it safely?

File: packages/ra-ui-materialui/src/button/DeleteWithConfirmButton.tsx [hunk 2]
```diff
@@ -34,6 +34,7 @@ export const DeleteWithConfirmButton = <RecordType extends RaRecord = any>(
         redirect = 'list',
         translateOptions = {},
         mutationOptions,
+        color = 'error',
```
Comment (slax57) [inline packages/ra-ui-materialui/src/button/DeleteWithConfirmButton.tsx:37]: There are linter warnings in this file (e.g. `'styled' is defined but never used`)

You should configure your IDE to show them directly.

File: packages/ra-ui-materialui/src/button/DeleteButton.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        <DeleteButton
+            mutationMode="pessimistic"
+            record={{ id: 1 }}
+            label="Delete"
+            resource="post"
+        />
+    </AdminContext>
+);
+
+export const WithUserDefinedPalette = () => (
```
Comment (slax57) [inline packages/ra-ui-materialui/src/button/DeleteButton.stories.tsx:41]: none of the stories actually test the case of the issue, along with the `color` props you changed. You should at least have one. E.g.:

```tsx
const theme2 = createTheme({
    palette: {
        primary: {
            light: colors.orange[100],
            main: colors.orange[500],
            contrastText: colors.grey[50],
        },
        error: {
            main: colors.orange[500],
        },
    },
});

export const ContainedWithUserDefinedPalette = () => (
    <AdminContext theme={theme2}>
        <DeleteButton
            variant="contained"
            color="primary"
            label="Delete"
            record={{ id: 1 }}
        />
    </AdminContext>
);
```

(with `colors` imported from MUI)

Comment (slax57) [inline packages/ra-ui-materialui/src/button/DeleteButton.stories.tsx:41]: none of the stories actually test the case of the issue, along with the `color` props you changed. You should at least have one. E.g.:

```tsx
const theme2 = createTheme({
    palette: {
        primary: {
            light: colors.orange[100],
            main: colors.orange[500],
            contrastText: colors.grey[50],
        },
        error: {
            main: colors.orange[500],
        },
    },
});

export const ContainedWithUserDefinedPalette = () => (
    <AdminContext theme={theme2}>
        <DeleteButton
            variant="contained"
            color="primary"
            label="Delete"
            record={{ id: 1 }}
        />
    </AdminContext>
);
```

(with `colors` imported from MUI)

**Review by slax57 — CHANGES_REQUESTED** [review]
Other than that, the code looks good IMO

**Review by slax57 — APPROVED** [review]

File: packages/ra-ui-materialui/src/button/DeleteWithConfirmButton.tsx (hunk 1 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/button/DeleteWithConfirmButton.tsx:149]: I think this is something we use to do for older versions of MUI

**Pull request merged**